ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java

```diff
@@ -218,23 +230,33 @@ private String getOsParam(String userAgentOs) {
   @VisibleForTesting
   List<Set<String>> loadAllowedUrls() throws IOException {
     if (singletonAllowedImpressionUrls == null || singletonAllowedClickUrls == null) {
-      singletonAllowedImpressionUrls = new HashSet<>();
-      singletonAllowedClickUrls = new HashSet<>();
+      Set<String> allowedImpressionUrls = new HashSet<>();
+      Set<String> allowedClickUrls = new HashSet<>();
 
       List<String[]> urlToActionTypePairs = readPairsFromFile(urlAllowList, "urlAllowList");
 
       // 0th element is site, 1st element is action type (click/impression)
       for (String[] urlToActionType : urlToActionTypePairs) {
         if (urlToActionType[1].equals("click")) {
-          singletonAllowedClickUrls.add(urlToActionType[0]);
+          allowedClickUrls.add(urlToActionType[0]);
         } else if (urlToActionType[1].equals("impression")) {
-          singletonAllowedImpressionUrls.add(urlToActionType[0]);
+          allowedImpressionUrls.add(urlToActionType[0]);
         } else {
           throw new IllegalArgumentException(
               "Invalid action type in url allow list: " + urlToActionType[1]);
         }
       }
+
+      singletonAllowedImpressionUrls = allowedImpressionUrls;
+      singletonAllowedClickUrls = allowedClickUrls;
     }
     return Arrays.asList(singletonAllowedClickUrls, singletonAllowedImpressionUrls);
   }
+
+  private void checkIfParamPresent(ParsedReportingUrl reportingUrl, String paramName) {
```

Comment:
The name here should make it more apparent that an exception will be thrown. Probably `requireParamPresent`.